salt/state.py

```diff
@@ -2507,22 +2508,16 @@ def _get_envs(self):
             envs.extend([x for x in list(self.opts['file_roots'])
                          if x not in envs])
         env_order = self.opts.get('env_order', [])
+        # Remove duplicates
+        env_order = list(collections.OrderedDict.fromkeys(env_order))
```

Comment:
I think that a variant on the last method which uses a set to store which are members and which are not, would be best:

```python
>>> import timeit
>>> prep = 'env_order = "[\'foo\', \'bar\', \'foo\', \'baz\']"'
>>> timeit.Timer('dummy = set(); [(v, dummy.add(v))[0] for v in env_order if v not in dummy]', prep).repeat()
[3.2361321449279785, 3.179112195968628, 3.1725778579711914]
>>> timeit.Timer('members = set(); env_order = [x for x in env_order if not (x in members or members.add(x))]', prep).repeat()
[1.925464153289795, 1.9767980575561523, 1.8665900230407715]
```

This is superior to the ``t5`` example, I think, because it avoids both the instantiation of a tuple, and the reference to the index of the tuple. It also avoids using ``set.add`` for each iteration.